File: frontend/lib/data/models/holiday_schedule.dart
import 'package:json_annotation/json_annotation.dart';

part 'holiday_schedule.g.dart';

@JsonSerializable()
class HolidaySchedule {
	final int id;
	final int year;
	final int? festivalId;
	final String? festivalName;
	final int? regionId;
	final String? regionName;
	final String startDate;
	final String endDate;
	final int totalDays;
	final String? workDays;
	final String? workDaysDesc;
	final String? description;
	final String? remark;
	final int status;

	HolidaySchedule({
		required this.id,
		required this.year,
		this.festivalId,
		this.festivalName,
		this.regionId,
		this.regionName,
		required this.startDate,
		required this.endDate,
		required this.totalDays,
		this.workDays,
		this.workDaysDesc,
		this.description,
		this.remark,
		this.status = 1,
	});

	factory HolidaySchedule.fromJson(Map<String, dynamic> json) =>
		_$HolidayScheduleFromJson(json);
	Map<String, dynamic> toJson() => _$HolidayScheduleToJson(this);
}

File: backend/src/main/java/com/regional/calendar/entity/HolidaySchedule.java
package com.regional.calendar.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("holiday_schedule")
public class HolidaySchedule {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Integer year;

    private Long festivalId;

    private String festivalName;

    private Long regionId;

    private String regionName;

    private LocalDate startDate;

    private LocalDate endDate;

    private Integer totalDays;

    private String workDays;

    private String workDaysDesc;

    private Integer isOfficial;

    private String announcement;

    private String source;

    private String sourceUrl;

    private String remark;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
